26_Blur/app_resources/shader.comp.hlsl

```diff
@@ -0,0 +1,221 @@
+        if (scanIdx > lastIdx) break;
+        
+        const float32_t leftIdx = scanIdx - radius;
+        const float32_t rightIdx = scanIdx + radius;
+
+        float32_t result = 0;
+        if (rightIdx <= lastIdx)
+        {
+            float32_t rightFloor, rightCeil;
+            smemAccessor.get(scanIdx + radiusFl, rightFloor);
+            smemAccessor.get(scanIdx + radiusCl, rightCeil);
+            result += lerp(rightFloor, rightCeil, alpha);
+        }
+        else switch (wrapMode)
+        {
+            case WRAP_MODE_CLAMP_TO_BORDER:
+            {
+                result += (rightIdx - lastIdx) * borderColor[channel];
+            } break;
+            case WRAP_MODE_CLAMP_TO_EDGE:
+            {
+                float32_t last, lastMinus1;
+                smemAccessor.get(lastIdx, last);
+                smemAccessor.get(lastIdx - 1, lastMinus1);
+                result += (rightIdx - lastIdx) * (last - lastMinus1);
+            } break;
+            case WRAP_MODE_REPEAT:
+            {
+                const uint16_t repeatedIdx = (uint16_t)(rightIdx % width);
+                float32_t repeatedValue;
+                smemAccessor.get(repeatedIdx, repeatedValue);
+                result += repeatedValue;
+            } break;
+            case WRAP_MODE_MIRROR:
+            {        
+                const uint16_t mirroredIdx = (uint16_t)((rightIdx / width) % 2 == 0 ? rightIdx % width : lastIdx - (rightIdx % width));
+                float32_t mirrored;
+                smemAccessor.get(mirroredIdx, mirrored);
+                result += mirrored;
+            } break;
+        }
+
+        if (leftIdx >= 0)
+        {
+            float32_t leftFloor, leftCeil;
+            smemAccessor.get(scanIdx - radiusFl, leftFloor);
+            smemAccessor.get(scanIdx - radiusCl, leftCeil);
+            result -= lerp(leftFloor, leftCeil, alpha);
+        }
+        else switch (wrapMode)
+        {
+            case WRAP_MODE_CLAMP_TO_BORDER:
+            {
+                result -= leftIdx * borderColor[channel];
+            } break;
+            case WRAP_MODE_CLAMP_TO_EDGE:
+            {
+                float32_t first;
+                smemAccessor.get(0, first);
+                result -= abs(leftIdx) * first;
+            } break;
+            case WRAP_MODE_REPEAT:
+            {
+                const uint16_t repeatedIdx = (uint16_t)((leftIdx % width + width) % width);
+                float32_t repeatedValue;
+                smemAccessor.get(repeatedIdx, repeatedValue);
+                result -= repeatedValue;
+            } break;
+            case WRAP_MODE_MIRROR:
+            {
+                const uint16_t mirroredIdx = (uint16_t)((-leftIdx / width) % 2 == 0 ? (-leftIdx) % width : lastIdx - ((-leftIdx) % width));
+                float32_t mirrored;
+                smemAccessor.get(mirroredIdx, mirrored);
+                result -= mirrored;
+            } break;
+        }
```

Comment:
why do you couple wrap offset calculation with fetching memory?

Reply:
this part is WIP and mostly broken but also note that `last`, `leftFloor`, etc aren't indices